tests/Doubles/FakeDownloader.php

```diff
@@ -9,8 +9,14 @@
 
 final class FakeDownloader implements Downloader
 {
+    // todo: remove this and allow only for in-memory content to explicit control
```

Comment:
yup, this class is to magical, while rewriting the tests, I was able to be surprised more than once, because the file system is not defined in the code